native/common/jp_array.cpp

```diff
@@ -71,14 +71,19 @@ JPPyObject JPArray::getRange(jsize start, jsize stop)
 void JPArray::setRange(jsize start, jsize stop, PyObject* val)
 {
 	JP_TRACE_IN("JPArray::setRange");
+
+	// Make sure it is an iterable before we start
+	if (!JPPySequence::check(val))
+		JP_RAISE_TYPE_ERROR("can only assign a sequence");
+
 	JPJavaFrame frame;
 	JPClass* compType = m_Class->getComponentType();
 	unsigned int len = stop - start;
 	JPPySequence seq(JPPyRef::_use, val);
-	size_t plength = seq.size();
+	long plength = seq.size();
```

Comment:
can size be negative or why change type here? If this was just added to avoid a compiler warning, I'd suggest to change JPPySequence size return type also to a size type (size_t).

Reply:
The problem here was that the error message reported a very large number instead of -1.  I will look at JPPySequence.

Reply:
Okay I confirmed it was already a jlong so it was supposed to be signed.  The ``size_t`` was incorrect.